Changes in module 'nf-core/bcftools/sort'
--- modules/nf-core/bcftools/sort/main.nf
+++ modules/nf-core/bcftools/sort/main.nf
@@ -11,8 +11,9 @@
     tuple val(meta), path(vcf)
 
     output:
-    tuple val(meta), path("*.{vcf,vcf.gz,bcf,bcf.gz}")  , emit: vcf
-    path "versions.yml"                                 , emit: versions
+    tuple val(meta), path("*.{vcf,vcf.gz,bcf,bcf.gz}"), emit: vcf
+    tuple val(meta), path("*.csi")                    , emit: csi, optional: true
+    path "versions.yml"                               , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
@@ -29,7 +30,7 @@
     """
     bcftools \\
         sort \\
-        --output ${prefix}.${extension} \\
+        --output ${prefix}.sorted.${extension} \\
         --temp-dir . \\
         $args \\
         $vcf
@@ -51,7 +52,7 @@
                     "vcf"
 
     """
-    touch ${prefix}.${extension}
+    touch ${prefix}.sorted.${extension}
 
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":

************************************************************
